### PR TITLE
fix: onboarding social signup layout jump

### PR DESCRIPTION
**Pull request by kopancek**
## Changes

- Added generic loader, based on [Figma design](https://www.figma.com/file/nmfWPS7x3kzLUvYMkBx2kW/daily.dev---Dev-Mode?node-id=5501%3A8388&mode=dev)
- Added storybook for generic loader
- Added `isLoading` state to the onboarding `auth`
- Set `isLoading` to `true` when 3rd party login provider is clicked
- Set `isLoading` to `false` when activeTab is changed in `AuthOptions`
- Display generic loader when `isLoading` is true and onboarding step is `Intro`

### Video

https://www.loom.com/share/9996e37d327443f59eb98926fcec807f?sid=2cf12457-039a-4f50-b8ce-cb189758de08

## Events

No new events

## Manual Testing

Tested in preview

### On those affected packages:
- [X] Have you done sanity checks in the webapp?
- [x] Have you done sanity checks in the extension?
- [ ] Does this not break anything in companion?

### Did you test the modified components media queries?
- [X] MobileL (420px)
- [X] Tablet (656px)
- [X] Laptop (1020px)

#### Did you test on actual mobile devices?
- [ ] iOS (Chrome and Safari)
- [ ] Android

MI-226 #done

### Preview domain
https://mi-226-fix-onboarding-social-signup.preview.app.daily.dev

### CODE REVIEW

**Review by nensidosari — APPROVED** [review]
Looks good

File: packages/shared/src/components/utilities/loaders/generic.tsx [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  label = 'Preparing...',
+}: Props): ReactElement => {
+  return (
+    <div className="fixed inset-0 z-modal flex items-center justify-center bg-background-default">
+      <div className="flex flex-col items-center gap-5">
+        <LoaderIcon
+          size={IconSize.XLarge}
+          className={classnames(
+            className,
+            'flex-shrink-0 animate-spin drop-shadow-[0_0_5px_#CE3DE3]',
```
Comment (rebelchris) [inline packages/shared/src/components/utilities/loaders/generic.tsx:22]: This arbitrary value doesn't seem right.
It's not in our design library or what's the story here?

**Review by rebelchris — COMMENTED** [review]
One minor comment :)

File: packages/shared/src/components/utilities/loaders/generic.tsx (hunk 1 above)
Comment (kopancek) [reply packages/shared/src/components/utilities/loaders/generic.tsx:22]: We have the color defined, it's mostly about convincing tailwind that it wants to use the CSS variable and not show it as transparent :/

Comment (kopancek) [reply packages/shared/src/components/utilities/loaders/generic.tsx]: Added a CSS variable for the shadow color and used that instead. Let me know what you think.

**Pull request merged**